pkg/autodiscovery/config_poller.go

```diff
@@ -50,14 +50,62 @@ func (pd *configPoller) stop() {
 }
 
 // start starts polling the provider descriptor
-func (pd *configPoller) start(ac *AutoConfig) {
+func (pd *configPoller) start(ctx context.Context, ac *AutoConfig) {
+	_, isStreaming := pd.provider.(providers.StreamingConfigProvider)
+
+	if !isStreaming {
+		pd.pollOnce(ctx, ac)
```

Comment:
could you explain why we need to run poolOnce() if not streaming? 
can we move it if the `else` before `go pd.poll(ac)` ?

Reply:
Two reasons:

1. Non-polling providers, like the `file` one. This polling used to be disguised outside of the config poller [here](https://github.com/DataDog/datadog-agent/blob/7.38.0/pkg/autodiscovery/autoconfig.go#L232-L271).
2. We need all configs to be loaded before AD's `LoadAndRun` returns, because of the `agent check` command.

We can't have in the else for reason number 1, `file` is not polling, so the `else` won't be reached (because of `if !pd.canPoll`)